src/konfo_indeksoija_service/kouta/indexer.clj

```diff
@@ -0,0 +1,62 @@
+(ns konfo-indeksoija-service.kouta.indexer
+  (:require [konfo-indeksoija-service.rest.kouta :as kouta-backend]
+            [konfo-indeksoija-service.kouta.koulutus :as koulutus]
+            [konfo-indeksoija-service.kouta.toteutus :as toteutus]
+            [konfo-indeksoija-service.kouta.haku :as haku]
+            [konfo-indeksoija-service.kouta.hakukohde :as hakukohde]
+            [konfo-indeksoija-service.util.time :refer [format-long-to-rfc1123]]
+            [konfo-indeksoija-service.elastic.docs :as docs]
+            [clojure.tools.logging :as log]))
+
+(defn index-koulutukset
+  [oids]
+  (if (> (count oids) 0)
+    (let [koulutukset (doall (pmap koulutus/create-index-entry oids))]
+      (docs/upsert-docs "koulutus-kouta" koulutukset))))
+
+(defn index-koulutus
+  [oid]
+  (index-koulutukset [oid]))
+
+(defn index-toteutukset
+  [oids]
+  (if (> (count oids) 0)
```

Comment:
`(if (seq oids) ...` tai koska ei o toista haaraa vois selkeämpi olla `(when-not (empty? seq) ...`